src/main/java/com/contentful/java/cma/CMAClient.java

```diff
@@ -321,33 +335,97 @@ public Builder setLogLevel(Logger.Level logLevel) {
+     * It might be used for internal tracking of Contentfuls tools.
+     *
+     * @param name    the name of the app.
+     * @param version the version in semver of the app.
+     * @return this builder for chaining.
+     */
+    public Builder setApplication(String name, String version) {
+      this.application = Section.app(name, parse(version));
+      return this;
+    }
+
+    /**
+     * Which integration is used.
+     *
+     * @param name    name of the integration.
+     * @param version the version of the integration.
+     * @return this builder for chaining.
+     */
+    public Builder setIntegration(String name, String version) {
+      this.integration = Section.integration(name, parse(version));
+      return this;
+    }
+
     /**
      * @return a {@link CMAClient} out of this {@link Builder}.
      */
     public CMAClient build() {
       return new CMAClient(this);
     }
 
+    /**
+     * @return default core call factory builder, used by the sdk.
+     */
     public OkHttpClient.Builder defaultCoreCallFactoryBuilder() {
       final OkHttpClient.Builder okBuilder = new OkHttpClient.Builder()
           .addInterceptor(new AuthorizationHeaderInterceptor(accessToken))
-          .addInterceptor(new UserAgentHeaderInterceptor(getUserAgent(propertiesReader)))
+          .addInterceptor(new UserAgentHeaderInterceptor(getUserAgent()))
+          .addInterceptor(new ContentfulUserAgentHeaderInterceptor(
+              createCustomHeaderSections(application, integration))
+          )
           .addInterceptor(new ContentTypeInterceptor(DEFAULT_CONTENT_TYPE))
           .addInterceptor(new ErrorInterceptor());
 
       return setLogger(okBuilder);
     }
 
+    /**
+     * @return default update api call factory builder, used by the sdk.
+     */
     public OkHttpClient.Builder defaultUploadCallFactoryBuilder() {
       final OkHttpClient.Builder okBuilder = new OkHttpClient.Builder()
           .addInterceptor(new AuthorizationHeaderInterceptor(accessToken))
-          .addInterceptor(new UserAgentHeaderInterceptor(getUserAgent(propertiesReader)))
+          .addInterceptor(new UserAgentHeaderInterceptor(getUserAgent()))
+          .addInterceptor(new ContentfulUserAgentHeaderInterceptor(
+              createCustomHeaderSections(application, integration))
+          )
           .addInterceptor(new ContentTypeInterceptor(OCTET_STREAM_CONTENT_TYPE))
           .addInterceptor(new ErrorInterceptor());
 
       return setLogger(okBuilder);
     }
 
+    Section[] createCustomHeaderSections(Section application, Section integration) {
+      final Properties properties = System.getProperties();
+
+      Section sdk;
+      try {
+        sdk = sdk("contentful-management.java", parse(getSDKVersionName()));
+      } catch (IOException e) {
+        sdk = sdk("contentful-management.java", new Version(0, 0, 0, "INVALID"));
+      }
+
+      return new Section[]{
+          sdk,
+          platform(
+              "java",
```

Comment:
According to the RFC is should be "Java". Either works for me, just update the wiki if you want to keep it lowercase.